Core/CategoryComparer.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Core.Categories;
+using System.Collections.Generic;
+
+namespace Core
+{
+    internal class CategoryComparer : IEqualityComparer<ICategory>
+    {
+        public bool Equals(ICategory x, ICategory y)
+        {
+            if (ReferenceEquals(x, y)) 
+                return true;
+
+            if (ReferenceEquals(x, null) || ReferenceEquals(y, null))
+                return false;
+
+            return x.Name == y.Name;
+        }
+
+        public int GetHashCode(ICategory category)
+        {
+            if (category is null) return 0;
+
+            return category.Name.GetHashCode();
+        }
+    }
+}
```

Comment:
Codacy found an issue: [Add a new line at the end of the file 'CategoryComparer.cs'.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6003477)